### PR TITLE
shared: Add RHEL.7.devel on s390x

### PR DESCRIPTION
**Pull request by ldoktor**
This adds the s390x variant of RHEL.7.devel.

If you are checking this on RHEL 7, note that you need kernel with KVM support enabled and you must use `nettype=user` because of https://bugzilla.redhat.com/show_bug.cgi?id=1505330

Also note that running this test from z/vm guest might result in many timeouts so it's advised to run directly from lpar.

### CODE REVIEW

File: shared/unattended/RHEL-7-series.ks [hunk 1]
```diff
@@ -49,6 +49,9 @@ prelink
 %end
 
 %post
+# FIXME: Remove the /dev/ttysclp0 workaround when s390x console bug is resolved
+# https://bugzilla.redhat.com/show_bug.cgi?id=1351968
+function ECHO { for TTY in `[ -e /dev/ttysclp0 ] && echo ttysclp0; cat /proc/consoles | cut -f1 -d' '`; do echo "$*" > /dev/$TTY; done }
 function ECHO { for TTY in `cat /proc/consoles | cut -f1 -d' '`; do echo "$*" > /dev/$TTY; done }
```
Comment (clebergnu) [inline shared/unattended/RHEL-7-series.ks]: You forgot to remove this implementation of `ECHO`.

File: shared/cfg/guest-os/Linux/RHEL.cfg [hunk 2]
```diff
@@ -20,3 +20,5 @@
             boot_path = ppc/ppc64
         x86_64, i386:
             kernel_params += " console=ttyS0,115200 console=tty0"
+        s390x:
+            kernel_params += " console=ttysclp0 debug ignore_loglevel"
```
Comment (clebergnu) [inline shared/cfg/guest-os/Linux/RHEL.cfg]: Is `debug` and `ignore_loglevel` meant to be permanent here? Either way, it'd be nice to explain the *why* for both `debug` and `ignore_loglevel` here.

File: shared/unattended/RHEL-7-series.ks (hunk 1 above)
Comment (ldoktor) [reply shared/unattended/RHEL-7-series.ks]: Oups, true...

File: shared/cfg/guest-os/Linux/RHEL.cfg (hunk 2 above)
Comment (ldoktor) [reply shared/cfg/guest-os/Linux/RHEL.cfg]: The same story as arm. As it's considered experimental (within Avocado-vt) I enabled debug in order to get more details during install. I can add the comment if you insist, but to me that looks unnecessary (there is no explanation of other params on other archs...)

**Comment by clebergnu** [comment]
OK, fair enough.

**Comment by ldoktor** [comment]
Removed the old `ECHO`, did not include the extra comment as no other options are described either.

**Comment by ldoktor** [comment]
Added 2 commits to set proper `virtio-scsi-ccw` ranges.

**Comment by ldoktor** [comment]
And yet another addition, the `-no-kvm-pit-reinjection` which is x86 specific only.

**Comment by ldoktor** [comment]
I just noticed the `pci_bus = None` is important for ccw device (I wrongly assumed it's None). Fixed in `virttest.qemu_devices: Correct virtio-scsi-ccw bus range` commit.

File: virttest/qemu_devices/qcontainer.py [hunk 3]
```diff
@@ -1389,7 +1389,7 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus,
             elif scsi_hba == 'virtio-scsi-device':
                 addr_spec = [256, 16384]
                 pci_bus = {'type': 'virtio-bus'}
-            if scsi_hba in ('spapr-vscsi', "virtio-scsi-ccw"):
+            elif scsi_hba in ('spapr-vscsi', "virtio-scsi-ccw"):
```
Comment (clebergnu) [inline virttest/qemu_devices/qcontainer.py]: Yes, makes sense.

File: virttest/qemu_devices/qcontainer.py [hunk 4]
```diff
@@ -1384,12 +1384,13 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus,
...
+            elif scsi_hba.startswith("virtio"):
                 addr_spec = [256, 16384]
-            elif scsi_hba == 'virtio-scsi-device':
-                addr_spec = [256, 16384]
-                pci_bus = {'type': 'virtio-bus'}
-            elif scsi_hba in ('spapr-vscsi', "virtio-scsi-ccw"):
+                if scsi_hba == 'virtio-scsi-device':
+                    pci_bus = {'type': 'virtio-bus'}
+                elif scsi_hba == 'virtio-scsi-ccw':
+                    pci_bus = None
```
Comment (clebergnu) [inline virttest/qemu_devices/qcontainer.py]: OK, the code matches the commit message.  I think I know the answer, but let me ask: why does `virtio-scsi-device` has a `pci_bus` while `virtio-scsi-ccw` does not?

File: shared/cfg/guest-os/Linux.cfg [hunk 5]
```diff
@@ -20,7 +20,7 @@
     cdrom_test_cmd = "dd if=%s of=/dev/null bs=1 count=1"
     cdrom_info_cmd = "cat /proc/sys/dev/cdrom/info"
     timedrift, timerdevice..boot_test:
-        !ppc64, ppc64le, aarch64:
+        i386, x86_64:
```
Comment (clebergnu) [inline shared/cfg/guest-os/Linux.cfg]: OK, makes sense.

**Comment by ldoktor** [comment]
OK, there are 2 acks, let me merge it...

**Pull request merged**